filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -48,25 +48,36 @@ type Harvester interface {
 	// Run is the event loop which reads from the source
 	// and forwards it to the publisher.
 	Run(input.Context, Source, Cursor, Publisher) error
+	// Monitor is required for detailed metrics.
+	Monitor(input.Context, Source)
```

Comment:
Instead of having a `Monitor` per harvester, I wonder if we should have the file watcher do the reporting. The file watcher would report the size update and we would find and update the actual state.